buildbot/buildbot/worker.py

```diff
@@ -375,10 +399,66 @@ def stage_hls(db, config):
             )
 
 
+def stage_afi(db, config):
+    """AFI stage: Create the AWS FPGA binary and AFI from the *.xclbin 
+    (Xilinx FPGA binary file)
+    """
+    with work(db, state.AFI_START, state.AFI, state.HLS_FINISH) as task:
+        # clean all the generated files from previous runs.
+        task.run(
+            ['sh', '-c', 'cd xclbin; \
+              rm -rf to_aws; rm -rf *afi_id.txt;\
+              rm -rf *.tar; rm -rf *afgi_id.txt'],
+            cwd=CODE_DIR
+        )
+        # find the hw synthesis xclbin file.
+        xclbin_file_path = task.run(
+            ['sh', '-c', 'find ./xclbin  -type f -iname "*hw\.*.xclbin"'],
+            cwd=CODE_DIR,
+            capture=True
+        )
+        xclbin_file = xclbin_file_path.stdout.decode(
+            'utf8').strip().replace('./xclbin/', '')
+        # generate AFI and aws binary.
+        task.run(
+            ['sh', '-c', 'cur=`pwd`; cd $AWS_FPGA_REPO_DIR ; \
+            source ./sdaccel_setup.sh > /dev/null; cd $cur/xclbin; \
+            $SDACCEL_DIR/tools/create_sdaccel_afi.sh \
+    	    -xclbin={} \
+            -s3_bucket={} \
+            -s3_dcp_key={} \
+            -s3_logs_key={}'.format(xclbin_file, config['S3_BUCKET'],
+                                 config['S3_DCP'], config['S3_LOG'])],
+            cwd=CODE_DIR
+        )
+        # loops every 5 minutes to check if the AFI is ready.
+        while True:
+            time.sleep(300)
+            afi_id_file = task.run(
+                ['sh', '-c', 'cat ./xclbin/*afi_id.txt'],
+                cwd=CODE_DIR,
+                capture=True
+            )
+            afi_json = json.loads(afi_id_file.stdout.decode('utf8').strip())
```

Comment:
Instead of using a shell to `cat` the file, you can read it directly, if you know the filename (which you might want to use glob for again):

```py
with open('./xclbin/foo_afi_id.txt') as f:
    afi_json = json.load(f)
```